### ISSUE TITLE
Application profiling

### ISSUE THREAD

**Issue by jw3** [ctc-oss/fapolicy-analyzer#350]
Provide ability to run arbitrary applications with fapolicyd in permissive debug mode while capturing the log for analysis.

1. Select "Run profiler" from menu
2. A dialog pops up to enter a command (free text, maybe a file picker)
3. Select "Start"
4. Stop fapolicyd daemon service (if it is running)
5. Start fapolicyd daemon in subprocess with permissive + debug modes, dumping output to temp file
6. When application is complete start the daemon service again (if it was running)
7. Parse and display the temp file from debug output in the analyzer view

Determining when the application is "complete" can be tricky in some cases, like with a GUI application that will open a window and may require interaction.  Interaction seems fine, just need to make sure the subprocess termination is detected in all cases so it doesn't hang.

The goal here is to make it easy to identify what entries are required for an application to function.

**Comment by tparchambault** [comment]
> Determining when the application is "complete" can be tricky in some cases, like with a GUI application that will open a window and may require interaction. Interaction seems fine, just need to make sure the subprocess termination

Agreed. This may not be possible in the general case for all applications that are comprised of many execution threads/processes/plug-ins paths e.g. different codecs loaded depending on the input data stream. Maybe setting expectations such that multiple iterations may be required to cover different use cases. Don't have a real solution other than to make it easy for the administrator to quickly alternate between user activities and analysis activities when performed manually. Internal execution automation or scripting might be something to consider down the road, allowing sequential runs of an application with different input data while capturing fapolicyd events for those runs.
